pet_code/scripts/tb_pet_validation.py

```diff
@@ -0,0 +1,271 @@
+                bad_mm.append([mm, key, len(check_ch_dict[mm][key]), idx])
+                print(f"{Fore.BLUE}PROBLEM! mm {mm} - {key} - only {len(check_ch_dict[mm][key])} ch detected{Style.RESET_ALL}")                
+                print(f"{Fore.BLUE}Colums/Rows (Slabs/Energy rows) {idx} (0 to 7){Style.RESET_ALL}")
+    return bad_mm
+
+
+
+if __name__ == '__main__':
+    args      = docopt(__doc__)
+    conf      = configparser.ConfigParser()
+    conf.read(args['--conf'])
+    start     = time.time()
+    map_file  = conf.get('mapping', 'map_file')#'pet_code/test_data/SM_mapping_corrected.yaml' # shouldn't be hardwired
+    infiles   = args['INFILE']
+    nsigma    = conf.getint('output', 'nsigma', fallback=2)
+    singles_flag = conf.get('filter', 'singles', fallback='False')
+
+    time_ch, eng_ch, mm_map, centroid_map, slab_map, tb_val_mapping = read_ymlmapping(map_file)
+    filt_type = conf.get('filter', 'type', fallback='Impacts')
+    # Should improve with an enum or something
+    if 'Impacts'  in filt_type:
+        min_chan   = tuple(map(int, conf.get('filter', 'min_channels').split(',')))
+        evt_select = filter_event_by_impacts(eng_ch, *min_chan, singles = singles_flag)
+    elif 'OneMod' in filt_type:
+        min_chan   = tuple(map(int, conf.get('filter', 'min_channels').split(',')))
+        evt_select = filter_impacts_one_minimod(eng_ch, *min_chan)
+    else:
+        print('No valid filter found, fallback to 4, 4 minimum energy channels')
+        evt_select = filter_event_by_impacts(eng_ch, 4, 4)
+
+    time_cal   = conf.get('calibration',   'time_channels', fallback='')
+    eng_cal    = conf.get('calibration', 'energy_channels', fallback='')
+    if time_cal or eng_cal:
+        cal_func = calibrate_energies(time_ch, eng_ch, time_cal, eng_cal)
+        out_cal  = "Cal"
+    else:
+        cal_func = lambda x: x
+        out_cal  = "WoCal"
+    end_r      = time.time()
+    for f_in in infiles:
+        out_dir = conf.get('output', 'out_dir')
+        png_dir = out_dir + "/pngs"
+        txt_dir = out_dir + "/txts"
+        if not os.path.isdir(out_dir):
+            os.makedirs(out_dir)                    
+            os.makedirs(png_dir)
+            os.makedirs(txt_dir)
+        
+        f_in_name       = f_in.split(os.sep)[-1]
+        f_in_name_split = f_in_name.split(".")
+        f_in_cal_name   = [".".join(f_in_name_split[0:-1]) + out_cal, f_in_name_split[-1]]  
+        out_base        = os.path.join(out_dir, ".".join(f_in_cal_name))      
+        out_base_png    = os.path.join(png_dir, ".".join(f_in_cal_name))
+        out_base_txt    = os.path.join(txt_dir, ".".join(f_in_cal_name))
+        
+        start           = time.time()
+        print("Reading file: {}...".format(f_in))
+        pet_reader      = read_petsys_filebyfile(f_in, mm_map, evt_select, singles = singles_flag)        
+        filtered_events = [cal_func(tpl) for tpl in pet_reader()]
```

Comment:
`read_petsys_filebyfile` se ha actualizado y es `pet_reader` que toma como argumento el nombre del fichero. Con esto puedes definir `pet_reader` antes de empezar el bucle y luego usas:
`filtered_events = [cal_func(tpl) for tpl in pet_reader(f_in)]`